src/test/credentials/provider/ssoCredentialProvider.test.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as sinon from 'sinon'
+import * as SDK from 'aws-sdk'
+import { SsoCredentialProvider } from '../../../credentials/providers/ssoCredentialProvider'
+import { SsoAccessTokenProvider } from '../../../credentials/sso/ssoAccessTokenProvider'
+import { DiskCache } from '../../../credentials/sso/diskCache'
+import { GetRoleCredentialsResponse } from 'aws-sdk/clients/sso'
+
+describe('SsoCredentialProvider', () => {
+    describe('refreshCredentials', () => {
+        let sandbox = sinon.createSandbox()
+
+        const ssoRegion = 'dummyRegion'
+        const ssoUrl = '123abc.com/start'
+        const ssoOidcClient = new SDK.SSOOIDC()
+        const cache = new DiskCache()
+        const ssoAccessTokenProvider = new SsoAccessTokenProvider(ssoRegion, ssoUrl, ssoOidcClient, cache)
+
+        const ssoAccount = 'dummyAccount'
+        const ssoRole = 'dummyRole'
+        const ssoClient = new SDK.SSO()
+        const sut = new SsoCredentialProvider(ssoAccount, ssoRole, ssoClient, ssoAccessTokenProvider)
+
+        const HOUR_IN_MS = 3600000
+        const validAccessToken = {
+            startUrl: ssoUrl,
+            region: ssoRegion,
+            accessToken: 'dummyAccessToken',
+            expiresAt: new Date(Date.now() + HOUR_IN_MS).toISOString(),
+        }
+
+        afterEach(() => {
+            sandbox.restore()
+        })
+
+        it('should invalidate cached access token if denied', async () => {
+            const stubAccessToken = sandbox.stub(ssoAccessTokenProvider, 'accessToken').resolves(validAccessToken)
+            const stubSsoClient = sandbox.stub(ssoClient, 'getRoleCredentials')
+            stubSsoClient.returns(({
+                promise: sandbox.stub().throws({ code: 'UnauthorizedException' }),
+            } as any) as SDK.Request<GetRoleCredentialsResponse, SDK.AWSError>)
+
+            const stubInvalidate = sandbox.stub(cache, 'invalidateAccessToken').returns()
+            try {
+                await sut.refreshCredentials()
+            } catch (err) {
+                assert.strictEqual(stubInvalidate.called, true)
```

Comment:
If refreshCredentials succeeds without throwing, this test would not fail. If you expect the call to fail, have a look at `assert.throws` or `src\test\shared\utilities\assertUtils.ts`. Have a look at any of your tests in this PR that have a try/catch to see if the testing logic needs adjusment.

Reply:
Removed all try/catches from tests.